File: javaapp/src/main/java/com/example/javaapp/DBHelper/DBHelper.java
package com.example.javaapp.DBHelper;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.example.javaapp.model.Category;
import com.readystatesoftware.sqliteasset.SQLiteAssetHelper;

import java.util.ArrayList;
import java.util.List;

//or mer sarqac bazayic ogtvinq
public class DBHelper extends SQLiteAssetHelper {
    //mer bazai hascene
    private static final String DB_NAME="EDMTQuiz2019.db";
    //mer bazai versione
    private static final int DB_VER=1;
    public static DBHelper instance;
    public static  synchronized DBHelper getInstance(Context context){
        if (instance==null)
instance=new DBHelper(context);
return instance;

    }
    public DBHelper(Context context) {
        super(context,DB_NAME,null, DB_VER);
    }
    //Get All Category
    public List<Category> getAllCategories(){
        SQLiteDatabase db=instance.getWritableDatabase();

        //SELECT column1, column2, ...
        //FROM table_name;
        Cursor cursor=db.rawQuery("SELECT *FROM Category;", null);
        List <Category> categories=new ArrayList<>();
        if (cursor.moveToFirst()){
            while (!cursor.isAfterLast()){
                Category category=new Category(cursor.getInt(cursor.getColumnIndex("ID")),
                        cursor.getString(cursor.getColumnIndex("Name")),
                        cursor.getString(cursor.getColumnIndex("Image")));
                categories.add(category);
                cursor.moveToNext();
            }
        }
        cursor.close();
        db.close();
        return categories;
    }

}

File: app/src/main/java/com/example/kotlinquizz/DBHelper/DBHelper.kt
package com.example.kotlinquizz.DBHelper

import android.content.Context
import com.example.kotlinquizz.model.Category
import com.example.kotlinquizz.model.Question
import com.readystatesoftware.sqliteasset.SQLiteAssetHelper

import kotlin.collections.ArrayList

class DBHelper(context: Context):SQLiteAssetHelper(context,DB_NAME,null,DB_VER) {
    companion object {
        private var instance: DBHelper? = null

        private val DB_NAME = "EDMTQuiz2019.db"
        private val DB_VER = 1
        @Synchronized
        fun getInstance(context: Context):DBHelper{
            if (instance==null)
                instance=DBHelper(context)
            return  instance!!
        }


    }
    //Get all Category
    val allCategory:List<Category>
    get(){
       val db= instance!!.writableDatabase
        val cursor=db.rawQuery("SELECT * FROM Category",null)
        val categoryies=ArrayList<Category>()
        if (cursor.moveToFirst()){
            while (!cursor.isAfterLast){
                val category=Category(cursor.getInt(cursor.getColumnIndex("ID")),
                    cursor.getString(cursor.getColumnIndex("Name")),
                    cursor.getString(cursor.getColumnIndex("Image")))
                categoryies.add(category)
                cursor.moveToNext()
            }
        }
        cursor.close()
        db.close()
        return  categoryies
    }
    //Get All question all category
    fun getQuestionByCategory(categoryId:Int):MutableList<Question>{
        val db= instance!!.writableDatabase
        val cursor=db.rawQuery("SELECT * FROM Question WHERE categoryId=$categoryId ORDER BY RANDOM() LIMIT 30",null)
        val questionList=ArrayList<Question>()
        if (cursor.moveToFirst()){
            while (!cursor.isAfterLast){
                val question=Question(cursor.getInt(cursor.getColumnIndex("ID")),
                    cursor.getString(cursor.getColumnIndex("QuestionText")),
                    cursor.getString(cursor.getColumnIndex("QuestionImage")),
                    cursor.getString(cursor.getColumnIndex("AnswerA")),
                    cursor.getString(cursor.getColumnIndex("AnswerB")),
                    cursor.getString(cursor.getColumnIndex("AnswerC")),
                    cursor.getString(cursor.getColumnIndex("AnswerD")),
                    cursor.getString(cursor.getColumnIndex("CorrectAnswer")),
                  if (cursor.getInt(cursor.getColumnIndex("IsImageQustion"))==0)  java.lang.Boolean.FALSE
                  else java.lang.Boolean.TRUE,cursor.getInt(cursor.getColumnIndex("CategoryID")))
questionList.add(question)
                cursor.moveToNext()
            }
        }
        cursor.close()
        db.close()
        return questionList
    }
    }
